explainer.md

```diff
@@ -1,624 +1,397 @@
 
-```html
-<html>
-<head>
-  <style>
-    body {
-      background-color: blue;
-    }
-    .animated {
-      position: relative;
-      top: 300px;
-      width: 100px;
-      height: 100px;
-      background-color: red;
-    }
-  </style>
-</head>
-<body>
-  <div class="animated" sharedid="header-id" id="header">Shared Element</div>
-  <a href="new-document.html">Click Me</a>
-</body>
-</html>
-```
+The aim of this design is to allow for representations of both Page-A and Page-B to exist at the same time, without the usability, accessibility, or memory concerns of having both complete DOM trees alive.
 
-### New Document
+Here's the example that will be used to explain the design:
 
-```html
-<html>
-<head>
-  <style>
-    body {
-      background-color: lightblue;
-    }
-    .animated {
-      position: relative;
-      top: 100px;
-      width: 200px;
-      height: 200px;
-      background-color: green;
-    }
-
-    ::shared-container(header-id) {
-      animation: ::shared-container-header-id 1s ease-in;
-    }
-  </style>
-</head>
-<body>
-  <div class="animated" sharedid="header-id" id="header">Shared Element</div>
-</body>
-</html>
+<img alt="Page-A and Page-B" src="media/pages.png?raw=true">
+
+The transition between Page-A and Page-B can be a full cross-document navigation between two same-origin pages, or it can be an SPA navigation. The main mechanism and concepts are the same between the two.
+
+Cross-origin transitions are something we want to tackle, but may have significant differences and restrictions for security reasons. Cross-origin transitions are not covered in this document.
+
+## Part 1: The offering
+
+Before Page-A goes away, it offers up parts of itself to be used in the transition. Generally, this will mean one part per 'thing' that acts independently during the transition. For the example transition, the parts are:
+
+- The header
+- The share button
+- The rest (referred to as the page root)
+
+https://user-images.githubusercontent.com/93594/141104275-6d1fb67a-2f73-41e4-9cef-14676798223b.mp4
+
+Aside from the root, an element can only be offered as a transition part if it has paint containment, which clips child content to the content box, although this can be expanded with `overflow-clip-margin`. The is a compromise made to greatly simplify implementation.
+
+Additionally, the element must be a single rect. As in, it doesn't break across lines or columns.
+
+- Open question: If the developer tries to break the above rules, what should happen? Is paint containment applied by the browser? Is that element dropped from the transition? Is the whole transition abandoned?
```

Comment:
Yeah, I think that's what I'm saying. I mean out of the two choices:

* If we require paint containment, it means that we require the developer to do some work before sharing the element. Then when the transition happens, we either 1. issue a warning if containment was not satisfied and skip it, or 2. ensure that element looks the same way during the transition as before
* If we automatically do paint containment, it means it will work automatically but 1. we still have to issue a warning if we can't do a containment (such as non-atomic inline - `<span>` for example) and 2. things may temporarily shift out of place and be clipped differently for the duration of the transition

I prefer the first option, but I guess that's why it's an open question :)